src/main/java/com/epam/edumanagementsystem/admin/rest/api/AcademicClassController.java

```diff
@@ -155,5 +155,34 @@ public String addNewAcademicCourseAndTeacher(@ModelAttribute("existingClass") Ac
         }
     }
 
+    @GetMapping("/{name}/classroom")
+    public String classroomTeacherForAcademicClass(@PathVariable("name") String name, Model model) {
+        AcademicClass academicClass = academicClassService.findByName(name);
+        model.addAttribute("teachers", academicClass.getTeacher());
+        model.addAttribute("existingClassroomTeacher", new AcademicClass());
+        if (academicClass.getClassroomTeacher() == null) {
+            return "classroomTeacherSection";
+        } else {
+            model.addAttribute("classroomTeacher", academicClass.getClassroomTeacher());
+            return "classroomTeacherSection";
+        }
+    }
+    @PostMapping("{name}/classroom")
+    public String addClassroomTeacherInAcademicClass(@ModelAttribute("existingClassroomTeacher") AcademicClass academicClass,
+                                                     @PathVariable("name") String name, Model model) {
+        AcademicClass academicClassFindByName = academicClassService.findByName(name);
+        model.addAttribute("teachers", academicClassFindByName.getTeacher());
+        Set<Teacher> allTeachersByAcademicClass = academicClassService.findAllTeachers(name);
```

Comment:
In the list of the teachers, from where we should chose the classroom teacher, there are also included teachers who are not teachers of that academic class.  4.Classroom teacher must be the teacher of that class - this requiremnt is violated.

Reply:
This was an unused field that should be deleted